### PR TITLE
Feature/eip 1283

### PR DESCRIPTION
**Pull request by zilm13**
Have not found any github tests for this, only unit provided

### CODE REVIEW

File: ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 1]
```diff
@@ -136,6 +137,7 @@ public Program(byte[] codeHash, byte[] ops, ProgramInvoke programInvoke, Transac
         traceListener = new ProgramTraceListener(config.vmTrace());
         this.memory = setupProgramListener(new Memory());
         this.stack = setupProgramListener(new Stack());
+        this.originalRepo = programInvoke.getRepository().getSnapshotTo(programInvoke.getRepository().getRoot());
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java]: Ooops, we could have not only `RepositoryImpl` here at least in tests.
Any ideas how can we get copy of Repo here?

File: ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 2]
```diff
@@ -789,10 +791,29 @@ private int getReturnDataBufferSizeI() {
                 Arrays.copyOfRange(returnDataBuffer, off.intValueSafe(), off.intValueSafe() + size.intValueSafe());
     }
 
+    @Deprecated
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java]: I'd not deprecate `storageLoad` cause `SLOAD` op should keep using it. `getCurrentValue` and `storageLoad` are different operations in terms of semantics, I'd decouple them from each other.

File: ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 3]
```diff
@@ -789,10 +791,29 @@ private int getReturnDataBufferSizeI() {
...
 
+    /**
+     * @return current Storage data for key
+     */
+    public DataWord getCurrentValue(DataWord key) {
+        return getStorage().getStorageValue(getOwnerAddress().getLast20Bytes(), key);
+    }
+
+    /*
+     * @return Storage data at the beginning of Program execution
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java]: Let's extend description with a reference to the EIP-1283

File: ethereumj-core/src/main/java/org/ethereum/core/Repository.java [hunk 4]
```diff
@@ -235,4 +235,6 @@ void loadAccount(byte[] addr, HashMap<ByteArrayWrapper, AccountState> cacheAccou
                      HashMap<ByteArrayWrapper, ContractDetails> cacheDetails);
 
     Repository getSnapshotTo(byte[] root);
+
+    Repository clone();
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/core/Repository.java]: Let's add a javadoc to highlight semantics of this call.

**Pull request merged**